File: app/src/main/java/com/mahmoudmabrok/hadibadi/Hady/HadyActivity.java
package com.mahmoudmabrok.hadibadi.Hady;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import androidx.recyclerview.widget.ItemTouchHelper;
import android.text.TextUtils;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

import com.mahmoudmabrok.hadibadi.Hady.Adapter.CustomAdapter;
import com.mahmoudmabrok.hadibadi.R;
import com.mahmoudmabrok.hadibadi.Util.Alert;
import com.mahmoudmabrok.hadibadi.Util.Operation;

import java.util.ArrayList;
import java.util.List;

public class HadyActivity extends AppCompatActivity {


    EditText edPlayer;
    private RecyclerView recyclerView;
    private CustomAdapter adapter;
    private List<String> listPlayer;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_hady);

        initView();

    }

    private void initView() {
        edPlayer = (EditText) findViewById(R.id.edHadyName);

        recyclerView = (RecyclerView) findViewById(R.id.rvHadyList);
        adapter = new CustomAdapter();
        LinearLayoutManager manager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(manager);
        recyclerView.setAdapter(adapter);

        listPlayer = new ArrayList<>();

        new ItemTouchHelper(new ItemTouchHelper.SimpleCallback(0,
                ItemTouchHelper.LEFT | ItemTouchHelper.RIGHT) {
            @Override
            public boolean onMove(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder, RecyclerView.ViewHolder target) {
                return false;
            }

            @Override
            public void onSwiped(RecyclerView.ViewHolder viewHolder, int direction) {
                int pos = viewHolder.getAdapterPosition();
                listPlayer.remove(pos);
                adapter.removeItem(pos);
            }
        }).attachToRecyclerView(recyclerView);
    }

    /**
     * select one player from list and draw it out
     * @param view
     */
    public void Select(View view) {
        int size = adapter.getItemCount();
        if (size > 1) {
            int rnd = Operation.getRandom(size);
            String playerName = adapter.drawOut(rnd);
            // display nam with dialog
            Alert.getDialog(HadyActivity.this, playerName).show();
        } else {
       //     showMessage("Add Player,(At least 2) ");
            edPlayer.setError("Add More Players");
        }
    }

    public void showMessage(String s) {
        Toast.makeText(this, s, Toast.LENGTH_LONG).show();
    }

    public void addPlayer(View view) {
        Operation.hideInputKeyboard(this);
        String s = edPlayer.getText().toString();
        if (!TextUtils.isEmpty(s)) {
            listPlayer.add(s);
            adapter.addItem(s);
            edPlayer.setText("");
        } else {
            edPlayer.setError("please add player name");
        }
    }
}

File: app/src/main/java/com/mahmoudmabrok/hadibadi/hady/HadyActivity.kt
package com.mahmoudmabrok.hadibadi.hady

import android.os.Bundle
import android.text.TextUtils
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.ItemTouchHelper
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.mahmoudmabrok.hadibadi.hady.adapter.CustomAdapter
import com.mahmoudmabrok.hadibadi.R
import com.mahmoudmabrok.hadibadi.util.Alert
import com.mahmoudmabrok.hadibadi.util.Operation
import kotlinx.android.synthetic.main.activity_hady.*

class HadyActivity : AppCompatActivity() {

    private val adapter: CustomAdapter by lazy { CustomAdapter() }
    private val listPlayer: MutableList<String> = mutableListOf()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_hady)
        initView()
    }

    private fun initView() {

        rvHadyList.adapter = adapter

        ItemTouchHelper(object : ItemTouchHelper.SimpleCallback(0,
                ItemTouchHelper.LEFT or ItemTouchHelper.RIGHT) {
            override fun onMove(recyclerView: RecyclerView, viewHolder: ViewHolder, target: ViewHolder): Boolean {
                return false
            }

            override fun onSwiped(viewHolder: ViewHolder, direction: Int) {
                val pos = viewHolder.adapterPosition
                listPlayer.removeAt(pos)
                adapter.removeItem(pos)
            }
        }).attachToRecyclerView(rvHadyList)
    }

    /**
     * select one player from list and draw it out
     * @param view
     */
    fun Select(view: View?) {
        val size = adapter.itemCount
        if (size > 1) {
            val rnd = Operation.getRandom(size)
            val playerName = adapter.drawOut(rnd)
            // display nam with dialog
            Alert.getDialog(this@HadyActivity, playerName).show()
        } else {
            edHadyName.error = "Add More Players"
        }
    }


    fun addPlayer(view: View?) {
        val s = edHadyName.text.toString()
        if (!TextUtils.isEmpty(s)) {
            listPlayer.add(s)
            adapter.addItem(s)
            edHadyName.setText("")
        } else {
            edHadyName.error = "please add player name"
        }
    }
}